### ISSUE TITLE
Error when loading completion from zsh

### ISSUE THREAD

**Issue by kbd** [click-contrib/click-completion#7]
Under zsh, [you run the following code](https://github.com/click-contrib/click-completion/blob/master/click_completion/zsh.j2):

```bash
if [[ "$(basename ${(%):-%x})" != "..." ]]; then
```

Under a login shell, the shell executable begins with a dash (eg. '-zsh'), so `basename` fails with:

```bash
$ eval "$(pipenv --completion)"
basename: illegal option -- z
usage: basename string [suffix]
       basename [-a] [-s suffix] string [...]
```

(`pipenv` uses `click_completion` internally.) To fix this, you can do:

```bash
basename -- ${(%):-%x}
```

Adding `--` stops anything starting with `-` from being parsed as an argument.

**Comment by glehmann** [comment]
looks good to me.
Could you submit a pull request with that fix?

**Comment by glehmann** [comment]
fixed in #10
